### DISCUSSION TITLE
Confirmed deletion of .cache/m2e, now hanging forever

### DISCUSSION THREAD

**Discussion by davidmichaelkarr** [eclipse-m2e/m2e-core#1192] [General]
I had installed 2022-12 a couple of weeks ago, but as I was still examining issues with it, when it asked me if I wanted to delete the ".cache/m2e" tree, I always clicked "Keep Cache".  This morning I decided it was time, so I clicked "Delete Cache".  That was over an hour ago now.  Eclipse has been hung since then.  I killed the process at one point, and restarted, but it's continuing to delete the cache and eventually hanging Eclipse.  At least, it's saying it's doing that. I've been checking the last modified date on files in the tree, and they aren't changing.

Can I kill the process again and just delete that tree from the filesystem?  Will that satisfy it?

I just checked that tree from a shell, and there's barely any files or used disk space left in that tree, but Eclipse is still hung. I see that one file is "..../min/write.lock".

**Comment by davidmichaelkarr** [comment]
I decided to rename the .cache/m2e directory (to m2e.bak), while Eclipse was shut down.  When I restarted it, it no longer had "Delete M2E Cache" (not sure what the wording was) in the status area, but it still hung Eclipse a few seconds after starting.  I am dead in the water here.

**Comment by davidmichaelkarr** [comment]
I am attempting to attach a thread dump, but it is presently ignoring my attempts to do so.

**Comment by davidmichaelkarr** [comment]
I posted a question about the same problem here: https://www.eclipse.org/forums/index.php/m/1856953/#msg_1856953 , and I was able to upload a thread dump there.

**Reply by HannesWell** [reply]
This is an annoying problem, but unfortunately I cannot see anything from m2e in the provided thread-dump. There is literally not a single hit if a search for `m2e`.
Furthermore I never experienced a similar behavior. Therefore I cannot really tell what the problem could be.

I only see some hits for `lsp4e`, the Language-Server integration for Eclipse. Do you see those unresponsiveness before you are opening an editor? Or only (shortly) after you opened one? The language-server would currently my only guess, but it is not an educated one.

**Comment by davidmichaelkarr** [comment]
I was able to reinstall Eclipse and create a new workspace and proceed to get it set up again.  It will take a while to reimport everything. Now that the .m2e/cache directory is gone, perhaps I won't run into that problem again.  I assume that by starting the process of deleting the cache, it put something into a state that deadlocks.

**Reply by davidmichaelkarr** [reply]
Unfortunately, the deadlock happened again.  I finished importing all of my repositories (quite a few), and not a minute after I started doing real work again. I got the hourglass cursor and it stopped responding.  I left it there for a while with no change.  For now, I'm retreating back to 2022-06, the last version I had no problems with this. I suppose I can try again, perhaps importing fewer projects, or closing most of them. I have no idea what the problem might be.
